### PR TITLE
Clean up `no-shadowed-variable`

### PR DESCRIPTION
**Pull request by andy-hanson**
#### PR checklist

- [ ] Addresses an existing issue: #0000
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update
- [X] Enable CircleCI for your fork (https://circleci.com/add-projects)

#### What changes did you make?

Some code cleanup in `no-shadowed-variable`.

### CODE REVIEW

File: src/rules/noShadowedVariableRule.ts [hunk 1]
```diff
@@ -158,6 +153,15 @@ class NoShadowedVariableWalker extends Lint.BlockScopeAwareRuleWalker<ScopeInfo,
     }
 }
 
-class ScopeInfo {
-    public variableNames: string[] = [];
+// ES6 Set shim for strings
+class Set {
```
Comment (ajafff) [inline src/rules/noShadowedVariableRule.ts]: Do we really need to shim ES6 features? Maybe we should just specify the correct libs for tsc

Comment (adidahiya) [inline src/rules/noShadowedVariableRule.ts]: Yeah, as per #1971, you can just use `Set` directly, please don't shim it.

**Comment by andy-hanson** [comment]
Waiting on #1984 to clean this up.

**Comment by nchen63** [comment]
@andy-hanson thanks!

**Pull request merged**